### PR TITLE
Supports for OSX 10.11.4

### PR DESCRIPTION
**Pull request by eccyan**
I build moon for using, but some linker errors occurred.
So, I fix those.

__My system environments__

- OSX 10.11.4(El Capitan)

__What I did__

- Uses glfw3
- Add dependented frameworks
  - IOKit
  - CoreVideo
  - Carbon
  - Cocoa

Can you review this?

### CODE REVIEW

File: mrb_config.rb [hunk 1]
```diff
@@ -113,7 +113,12 @@
     l.library_paths << File.expand_path('soil', bvd)
     l.library_paths.uniq!
 
-    l.libraries << 'glfw'
+    unless Platform.darwin?
+      l.libraries << 'glfw'
```
Comment (archSeer) [inline mrb_config.rb]: I think the new header is `'glfw3'` on all platforms now, so just:

```ruby
l.libraries << 'glfw3'
```

That should fix the Travis build

Comment (IceDragon200) [inline mrb_config.rb]: glfw3 is the static library, there is an issue in the cmake file, causing it to build only the static library

**Comment by eccyan** [comment]
Oops, I mistake commit..

File: mrb_config.rb (hunk 1 above)
Comment (eccyan) [inline mrb_config.rb]: Ok, I got it. But I cannot fix travis..

Comment (archSeer) [inline mrb_config.rb]: Ah sorry, your code is fine, @IceDragon200 just fixed travis in master branch. If you rebase against master, it should be fixed

**Comment by IceDragon200** [comment]
@eccyan Well, I've fixed the travis build, could you try pulling the changes and retrying your build, I'm arranging with archSeer to setup a macosx travis build

File: bin/bin [hunk 2]
```diff
@@ -0,0 +1 @@
+bin
```
Comment (IceDragon200) [inline bin/bin]: :cry: Can we have our bin directory back?

File: mrb_config.rb (hunk 1 above)
Comment (eccyan) [inline mrb_config.rb]: @IceDragon200 I do not know a little better. Do you mean there is a problem in the use of static files?
I 'm glad, if you fix this on behalf of me 🙏

File: bin/bin (hunk 2 above)
Comment (eccyan) [inline bin/bin]: I have mistake 😢 Ok, I'll fix it tomorrow.
Because I'm half asleep . now time is 0:47 in Japan. I need to sleep soon . .

Comment (archSeer) [inline bin/bin]: No worries, we are happy to receive a PR from new contributors!
Same here, I currently live in Tokyo :)

Comment (eccyan) [inline bin/bin]: @archSeer Oh, NICE ! I live in Tokyo , too 😃

**Comment by eccyan** [comment]
Good morning ☀️  I fixed `bin` is gone.

**Comment by eccyan** [comment]
@IceDragon200 I was sure that your commit(bcda5a91bcd4d719a9f2753c8d502d43605ffdc8) can build.

**Comment by eccyan** [comment]
```
$ brew install cmake libyaml glfw3 glew
$ bundle exec rake build
...
[ 96%] Built target moon-mruby
(in /Users/sato/devel/moon)
Files:          41
Modules:         6 (    4 undocumented)
Classes:        25 (   14 undocumented)
Constants:      20 (   15 undocumented)
Methods:       290 (   61 undocumented)
 72.43% documented
[ 96%] Built target moon-docs
[100%] Built target makefont
```

Thanks supports 👍  I'll close this pr.

**Pull request closed without merging**